app/logic/minor.py

```diff
@@ -65,14 +65,49 @@ def getMinorProgress():
                             'hasSummer': "Completed" if student.hasSummer else "Incomplete"} for student in engagedStudentsWithCount]
     return engagedStudentsList
 
-def toggleMinorInterest(username):
+def toggleMinorInterest(username, isAdding):
     """
         Given a username, update their minor interest and minor status.
     """
-    user = User.get(username=username)
-    user.minorInterest = not user.minorInterest
+    
+    try:
+        user = User.get(username=username)
+        if not user:
+            return {"error": "User not found"}, 404
+        
+        user.minorInterest = isAdding
+        user.declaredMinor = False
+        user.save()
+
+    except Exception as e:
+        print(f"Error updating minor interest: {e}")
+        return {"error": str(e)}, 500
+    
+def declareMinorInterest(username):
+    """
+    Given a username, update their minor declaration
+    """
+    user = User.get_by_id(username)
+    
+    if not user:
+        raise ValueError(f"User with username '{username}' not found.")
+    
+    user.declaredMinor = not user.declaredMinor
+    
+    try:
+        user.save()
+    except Exception as e:
+        raise RuntimeError(f"Failed to declare interested student: {e}")
+    
+def getDeclaredMinorStudents():
+    """
+    Get a list of the students who have declared minor
+    """
+    declaredStudents = (User.select(User).where(User.isStudent & User.minorInterest & User.declaredMinor))
```

Comment:
Feels like we don't need to specify User inside the select statement, right? The parenthesis are probably also a little unnecessary.